lib/src/components/dropdown/dropdown.dart

```diff
@@ -218,6 +237,20 @@ class _ZetaDropDownState<T> extends State<ZetaDropdown<T>> {
   double get _size => widget.size == ZetaDropdownSize.mini ? 120 : 320;
 
   void onTap() {
+    /// Version 1 : Calculate if overflow happens based on using calculations from sizes.
+    final height = MediaQuery.of(context).size.height;
+    final headerRenderBox = _headerKey.currentContext!.findRenderObject()! as RenderBox;
+    final dropdownItemHeight = headerRenderBox.size.height;
+
+    /// Calculate if overflow can happen
+    final headerPosY = _headerPos.dy;
+
+    if (headerPosY + (dropdownItemHeight * (widget.items.length + 1)) > height) {
```

Comment:
I think here you should also have an else, to reset it to going down in the event that the user scrolls down the page

```dart
 } else {
      setState(() {
        _menuPosition = MenuPosition.down;
      });
```